### PR TITLE
Include specials when queuing subsequent episodes

### PR DESCRIPTION
**Pull request by linetrimmer**
**Changes**

- Include specials when queuing subsequent episodes if the server is set to display specials within the seasons they aired in
- Continues playing the next season after finishing the last episode of a season

### CODE REVIEW

**Comment by ManuLinares** [comment]
Interesting, gonna test this later.

**Review by nielsvanvelzen — COMMENTED** [review]
Everything seems to be working as expected. One thing I'm worried about though is that the next up screen does not show the season name. This can be confusing when going from a season 1 -> season 2.

**Comment by linetrimmer** [comment]
It could be something like 
S1E9. The Title Titles
S3E19-20. The Title Returns
Special - The Titling

I can add these to the next up screen PR if you think they're good

File: app/src/main/java/org/jellyfin/androidtv/util/apiclient/PlaybackHelper.java [hunk 1]
```diff
@@ -58,27 +59,29 @@ public static void getItemsToPlay(final BaseItemDto mainItem, boolean allowIntro
...
-                                        if (item.getIndexNumber() > mainItem.getIndexNumber()) {
+                                        if (foundMainItem) {
                                             if (!LocationType.Virtual.equals(item.getLocationType())) {
                                                 items.add(item);
                                             } else {
                                                 //stop adding when we hit a missing one
                                                 break;
                                             }
+                                        } else if (item.getId() != null && item.getId().equals(mainItem.getId())) {
+                                            foundMainItem = true;
```
Comment (thornbill) [inline app/src/main/java/org/jellyfin/androidtv/util/apiclient/PlaybackHelper.java:76]: This _should_ be doable with the api by setting the `StartItemId` but it seems that is not supported in the old apiclient. It maybe a good idea to add a TODO or FIXME comment here about it though.

**Review by thornbill — CHANGES_REQUESTED** [review]

**Comment by linetrimmer** [comment]
Added the TODO for StartItemId, a check for the current user being null, and a limit for how many episodes get queued.

Went with "S{x}:E{x} - Title" and "Special - Title" for the next up screen in #850. Even without queuing subsequent seasons, it can be confusing when playing episodes by Shuffle All since the queued episodes can jump from season to season.

File: app/src/main/java/org/jellyfin/androidtv/util/apiclient/PlaybackHelper.java [hunk 2]
```diff
@@ -63,13 +63,18 @@ public static void getItemsToPlay(final BaseItemDto mainItem, boolean allowIntro
                             @Override
                             public void onResponse(ItemsResult response) {
                                 if (response.getTotalRecordCount() > 0) {
+                                    // TODO: Finding the main item should be possible in the query using StartItemId, but it is not currently supported.
+                                    // With StartItemId added, the limit could also be included in the query.
                                     boolean foundMainItem = false;
+                                    final int limit = 150; // guard against too many items - account for episodes shorter than 5 min
```
Comment (thornbill) [inline app/src/main/java/org/jellyfin/androidtv/util/apiclient/PlaybackHelper.java:69]: This limit should probably be set as a class level constant.

**Review by thornbill — CHANGES_REQUESTED** [review]

**Review by thornbill — APPROVED** [review]

**Pull request merged**